ui/jobs/jobs.js

```diff
@@ -2381,12 +2385,102 @@ class Bars {
   }
 
   setupWhm() {
+    const lilyBox = this.addResourceBox({
+      classList: ['whm-color-lily'],
+    });
+    const lilysecondBox = this.addResourceBox({
+      classList: ['whm-color-lilysecond'],
+    });
+
+    const diaBox = this.addProcBox({
+      id: 'whm-procs-dia',
+      fgColor: 'whm-color-dia',
+    });
+    const assizeBox = this.addProcBox({
+      id: 'whm-procs-assize',
+      fgColor: 'whm-color-assize',
+    });
+    const lucidBox = this.addProcBox({
+      id: 'whm-procs-lucid',
+      fgColor: 'whm-color-lucid',
+    });
+
+    // BloodLily Guage
+    const stacksContainer = document.createElement('div');
+    stacksContainer.id = 'whm-stacks';
+    this.addJobBarContainer().appendChild(stacksContainer);
+    const bloodlilyContainer = document.createElement('div');
+    bloodlilyContainer.id = 'whm-stacks-bloodlily';
```

Comment:
I think my one nitpick with this UI is that the bloodlily boxes don't line up with the hp and mp bar.  I think this is also true of the smn ruin4 stacks.  It'd be nice to make them all the same width, maybe using flexbox with `justify-content: space-between` to spread them out to the right length.  I think in this case (and the smn case), I am not sure why `whm-stacks-bloodlily` needs to exist.  It seems like all of the stacks could be added directly to `whm-stacks`? It's an improvement to think about for the future.